packages/typespec-client-generator-core/src/types.ts

```diff
@@ -1704,9 +1705,9 @@ function handleServiceOrphanType(context: TCGCContext, type: Model | Enum | Unio
     const sdkType = diagnostics.pipe(getClientTypeWithDiagnostics(context, type));
     updateUsageOfModel(context, UsageFlags.Input | UsageFlags.Output, sdkType);
   }
-  if (getAccessOverride(context, type) !== undefined) {
-    const sdkType = diagnostics.pipe(getClientTypeWithDiagnostics(context, type));
-    updateUsageOfModel(context, UsageFlags.None, sdkType);
+  const sdkType = diagnostics.pipe(getClientTypeWithDiagnostics(context, type));
+  if ((sdkType.kind === "model" || sdkType.kind === "enum") && sdkType.usage === UsageFlags.None) {
+    updateUsageOfModel(context, getDefaultUsage(context, type.namespace), sdkType);
```

Comment:
```suggestion
  const sdkType = diagnostics.pipe(getClientTypeWithDiagnostics(context, type));
  updateUsageOfModel(context, UsageFlags.None, sdkType);
```

all logic could be covered by `getUsageOverride`.

Reply:
also, if not put the ns override into `getUsageOverride`, then only orphan model will take `@usage` on ns.